### PR TITLE
[Admin Portal Revamp] Microgateway Labels issue fixes

### PR DESCRIPTION
**Pull request by chathuranga95**
### Changes:
- Unnecessary `new Promise()` removal.

- Validation improvements. Now using `hapi Joi`.
mg labels validations

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx [hunk 1]
```diff
@@ -73,8 +79,11 @@ const InputList = (props) => {
                                     onChange={handleInput}
                                     label={inputLabelPrefix + ' ' + labelCounter++}
                                     value={item.value}
-                                    helperText={helperText}
+                                    helperText={
+                                        userInputItems.filter((obj) => obj.key === item.key)[0].error || helperText
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx:83]: Let's do a find here, Filter will iterate over all the values in the array where find will break in the fist finding , and won't need to use `[0]`

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx [hunk 2]
```diff
@@ -73,8 +79,11 @@ const InputList = (props) => {
...
+                                    }
                                     variant='outlined'
+                                    error={userInputItems.filter((obj) => obj.key === item.key)[0].error}
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx:86]: Same as above , Let's use `.find` here too

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx (hunk 1 above)
Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx]: Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8535/commits/5ea6a84b8d1a47454611ce6d7f6cd7bac359513f

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx (hunk 2 above)
Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx]: Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8535/commits/5ea6a84b8d1a47454611ce6d7f6cd7bac359513f

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/MicrogatewayLabels/AddEditMGLabel.jsx [hunk 3]
```diff
@@ -86,6 +87,23 @@ function AddEditMGLabel(props) {
...
+    const handleHostValidation = (hostName) => {
+        const schema = Joi.string().uri().empty();
+        const validationError = schema.validate(hostName).error;
+
+        if (validationError) {
+            const errorType = validationError.details[0].type;
+            if (errorType === 'any.empty') {
+                return 'Host is empty';
+            }
+            if (errorType === 'string.uri') {
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/MicrogatewayLabels/AddEditMGLabel.jsx:100]: we can do this in an elseif() right? Do we need 2 if conditions?

Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/MicrogatewayLabels/AddEditMGLabel.jsx]: Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8535/commits/12e3962cb2b86c5983390a9d137c7184595b0cd5

**Pull request merged**